examples/5-steps-with-keras.ipynb

```diff
@@ -0,0 +1,522 @@
+    "        self.datagen = joblib.load(filepath)\n",
+    "        return self"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can put together the first steps of our pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reshape_step = Step(\n",
+    "    name=\"reshape\",\n",
+    "    transformer=ReshapeData(),\n",
+    "    input_data=['input'],\n",
+    "    adapter={\n",
+    "        'X': [('input', 'images')],\n",
+    "        'y': [('input', 'labels')]\n",
+    "    },\n",
+    "    cache_dirpath=CACHE_DIR\n",
+    ")\n",
+    "\n",
+    "loader_step = Step(\n",
+    "    name=\"loader\",\n",
+    "    transformer=PrepareLoader(),\n",
+    "    input_steps=[reshape_step],\n",
+    "    cache_dirpath=CACHE_DIR\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To check that what we did actually works, let's define an auxilliary step that displays the generated image data stream."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class DataDisplay(BaseTransformer):\n",
+    "    def transform(self, datagen, X, y, **kwargs):\n",
+    "        img_batch, lbl_batch = datagen.flow(X, y, batch_size=BATCH_SIZE).next()\n",
+    "        n_row = BATCH_SIZE // 8\n",
+    "        fix, axs = plt.subplots(n_row, 8, figsize=(8, 2 * n_row))\n",
+    "        for i, ax in enumerate(axs.ravel()):\n",
+    "            ax.imshow(img_batch[i].reshape(8, 8), cmap='gray')\n",
+    "            ax.axis('off')\n",
+    "            ax.set_title('lbl = {}'.format(lbl_batch[i]))\n",
+    "            \n",
+    "display_step = Step(\n",
+    "    name=\"display\",\n",
+    "    transformer=DataDisplay(),\n",
+    "    input_steps=[reshape_step, loader_step],\n",
+    "    adapter={\n",
+    "        'datagen': [(loader_step.name, 'datagen')],\n",
+    "        'X': [(reshape_step.name, 'X')],\n",
+    "        'y': [(reshape_step.name, 'y')]\n",
+    "    },\n",
+    "    cache_dirpath=CACHE_DIR\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "display_step"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "display_step.fit_transform(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Steps for CNN training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We proceed to the crux of this notebook: step/transformer wrapping a Keras model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class KerasCnn(BaseTransformer):\n",
```

Comment:
@grzes314 We already have KerasModelTransformer in steps.keras.models and I would use just that. Also I think (sorry if not communicated properly) that one of the purposes of this notebook is to show that there is a toolkit already with all the boilerplate prepared.